### ISSUE TITLE
Alt-text tag not being preserved by quarto for inline images

### ISSUE THREAD

**Issue by dhodge180** [quarto-dev/quarto-cli#9535]
### Bug description

I'm having trouble with getting alt-text working through quarto. I am following the guidance in the manual but it doesn't appear to work. Compiling in RStudio as an Rmd file using knitr works fine, but using quarto to compile the same file loses the fig-alt, id and fig-align tags. The manual is here: https://quarto.org/docs/authoring/figures.html#alt-text

Am I just doing something silly, or is this a bug? Thanks.

### Steps to reproduce

Minimal example: running `quarto render file.qmd --to html` on this file

```
---
title: "Alt-text testing"
output: html_document
date: "2024-04-30"
engine: knitr
---

![My caption](twit.png "This is the HTML title tag"){#myid fig-alt="This gets lost" fig-align="right" width="420px"}

Compiling using `Render` in RStudio with knitr engine works fine. It wraps the image in a div which receives the id. (i.e. not using quarto, just RMarkdown to Pandoc engine I guess). And copies across the classes/attributes passed. (So you actually need to write just alt="Alt stuff" and not fig-alt="Alt stuff")

But compiling using `quarto preview`:

+ The caption works.
+ The title tag works.
+ The width tag works.
+ The id tag gets lost.
+ The fig-alt gets lost.
+ The fig-align is correctly applied via a quarto class to the container div.

I may just be doing something wrong as the manual says this should work fine.
```

### Expected behavior

I would expect the output html (from quarto rendering) to have a caption, title, id, alt, etc.. tags. But some of the tags are lost.

### Actual behavior

This is the html of what is created:

```
<div class="quarto-figure quarto-figure-right">
<figure class="figure">
<p><img src="twit.png" title="This is the HTML title tag" class="img-fluid figure-img" width="420"></p>
<figcaption>My caption</figcaption>
</figure>
</div>
```

### Your environment

RStudio 23.12.1 and also VSCode 1.88.1
(In Fedora, and also Windows)

### Quarto check output

```
Quarto 1.4.553
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.553
      Path: /opt/quarto-1.4.553/bin

[✓] Checking tools....................OK
      Chromium: 869685
      TinyTeX: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/bin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /usr/lib64/R
      LibPaths:
        - /home/david/R/x86_64-redhat-linux-gnu-library/4.3
        - /usr/lib64/R/library
        - /usr/share/R/library
      knitr: 1.46
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK
```

**Comment by cscheid** [comment]
I believe this has been fixed in the 1.5 prereleases:

<img width="1703" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/69e59dc8-965b-4873-8428-166c018b784f">

**Comment by cderv** [comment]
With latest quarto 1.5 I get the correct HTML I guess 

````html
<div class="quarto-figure quarto-figure-right">
<figure class="figure">
<p><img src="twit.png" title="This is the HTML title tag" class="img-fluid figure-img" alt="This gets lost" width="420"></p>
<figcaption>My caption</figcaption>
</figure>
````

WIth Quarto 1.4 it is not there I can confirm 

````html
<div class="quarto-figure quarto-figure-right">
<figure class="figure">
<p><img src="twit.png" title="This is the HTML title tag" class="img-fluid figure-img" width="420"></p>
<figcaption>My caption</figcaption>
</figure>
</div>
````

So somehow, this is already fixed. 

With `1.3` this is also there 
````html
<div id="myid" class="quarto-figure quarto-figure-right anchored">
<figure class="figure">
<p><img src="twit.png" title="This is the HTML title tag" class="img-fluid figure-img" alt="This gets lost" width="420"></p>
<figcaption class="figure-caption">My caption</figcaption>
</figure>
</div>
````

Not that in 1.3 we got the `anchored` class being set and id being preserved

So yes it is fixed, but we had regression along the way between 1.3 and 1.4. 

I can't find the exact commit that fixed this in 1.5 (would need to bisect), but wondering if we should consider this regression to backport ? (if we can find a scoped fix) 

1.5 release is soonish - so maybe this can wait

**Comment by cderv** [comment]
> I can't find the exact commit that fixed this in 1.5 (would need to bisect), but wondering if we should consider this regression to backport ? (if we can find a scoped fix)

In fact it could be 
- https://github.com/quarto-dev/quarto-cli/issues/8785 backported to 1.4

but the follow up  regression fix has not been backported 
- https://github.com/quarto-dev/quarto-cli/issues/9190 

This is file in 1.4 
https://github.com/quarto-dev/quarto-cli/blob/96e570774ee5b4fc2088f5560ac40abf414119de/src/resources/filters/layout/html.lua#L170-L175

and the fix from #9190 (https://github.com/quarto-dev/quarto-cli/commit/cbf6012c4223d45a210220b9222b605e2be238df) 
https://github.com/quarto-dev/quarto-cli/blob/0b7ed68355cd44888dad3c54c7954ca9225ba4a3/src/resources/filters/layout/html.lua#L170-L175

So @cscheid I would argue we should backport #9190 also (commit cbf6012c4223d45a210220b9222b605e2be238df)

Does my investigation make sense ?

**Comment by cscheid** [comment]
It does. I'm happy with backporting, and if you do it today (and it passes the v1.4 test suite), it'll make it into the patch release tomorrow :)
